### PR TITLE
Replace !isnan with isfinite

### PR DESCRIPTION
**Pull request by t-bltg**
Fix https://github.com/JuliaPlots/UnicodePlots.jl/issues/106.

### CODE REVIEW

File: src/interface/lineplot.jl [hunk 1]
```diff
@@ -101,8 +101,7 @@ function lineplot(
         color::UserColorType = :auto,
         name = "",
         kw...)
-    idx = map(!isnan, x)
-    idx .&= map(!isnan, y)
+    idx = map(isfinite, x) .& map(isfinite, y)
```
Comment (johnnychen94) [inline src/interface/lineplot.jl:104]: This doesn't matter much but I'm sensitive to unnecessary memory allocations :)

```suggestion
    idx = map(x, y) do i,j
        isfinite(i) && isfinite(j)
    end
```

**Comment by johnnychen94** [comment]
I hope we could always have a testset for every non-trivial PR so that we don't meet similar issues in the future.

File: src/interface/lineplot.jl (hunk 1 above)
Comment (t-bltg) [reply src/interface/lineplot.jl]: You're absolutely right !

**Comment by t-bltg** [comment]
> I hope we could always have a testset for every non-trivial PR so that we don't meet similar issues in the future.

Will add, once I fix https://github.com/JuliaPlots/UnicodePlots.jl/pull/151, which turned to be non trivial to fix.

**Comment by t-bltg** [comment]
Damn, the new map block now fails the CI :confused:

**Comment by t-bltg** [comment]
Added a test in `tst_canvas.jl` and `tst_lineplot.jl`.

**Review by johnnychen94 — APPROVED** [review]

**Comment by johnnychen94** [comment]
Evizero enjoys writing tests (and sometimes too verbose) so I think we should keep his good habit 😄

**Pull request merged**